File: Kotlin/Messenger/app/src/main/java/io/chirp/messenger/TransactionStatus.java
package io.chirp.messenger;

public enum TransactionStatus {
    _00 {
        @Override
        String description() {
            return "Transaction Failed";
        }
    },
    _01 {
        @Override
        String description() {
            return "Transaction successful";
        }
    },
    _09 {
        @Override
        String description() {
            return "Insufficient Funds";
        }
    };

    abstract String description();
}

File: Kotlin/Messenger/app/src/main/java/io/chirp/messenger/TransactionStatus.kt
package io.chirp.messenger

enum class TransactionStatus {
    _00 {
        override fun description(): String {
            return "Transaction Failed"
        }
    },
    _01 {
        override fun description(): String {
            return "Transaction successful"
        }
    },
    _09 {
        override fun description(): String {
            return "Insufficient Funds"
        }
    };

    internal abstract fun description(): String
}
